models/autoregressive-v1/model.jl

```diff
@@ -0,0 +1,101 @@
+using RxInfer, LinearAlgebra
+
+function initial_state(arguments)
+    return Dict(
+        "order" => arguments["order"],
+        "horizon" => arguments["horizon"],
+        "x_μ" => zeros(arguments["order"]),
+        "x_Λ" => diageye(arguments["order"])
+    )
+end
+
+function initial_parameters(arguments)
+    return Dict(
+        "τ_α" => 1.0, "τ_β" => 1.0, "θ_μ" => zeros(arguments["order"]), "θ_Λ" => 1e-10*diageye(arguments["order"])
+    )
+end
+
+@model function AR_model(y, order, parameters, state)
+    # `c` is a unit vector of size `order` with first element equal to 1
+    c = ReactiveMP.ar_unit(Multivariate, order)
+
+    τ ~ Gamma(α = parameters["τ_α"], β = parameters["τ_β"])
+    θ ~ MvNormal(mean = parameters["θ_μ"], precision = parameters["θ_Λ"])
+    x0 ~ MvNormal(mean = state["x_μ"], precision = state["x_Λ"])
+
+    x_prev = x0
+
+    for i in eachindex(y)
+        x[i] ~ AR(x_prev, θ, τ)
+        y[i] ~ Normal(mean = dot(c, x[i]), precision = 1e10)
+
+        x_prev = x[i]
+    end
+end
+
+@constraints function AR_constraints()
+    q(x0, x, θ, τ, y) = q(x0, x)q(y)q(θ)q(τ)
+end
+
+@meta function AR_meta(order)
+    AR() -> ARMeta(Multivariate, order, ARsafe())
+end
+
+@initialization function AR_init(parameters)
+    q(τ) = GammaShapeRate(parameters["τ_α"], parameters["τ_β"])
+    q(θ) = MvNormalMeanPrecision(parameters["θ_μ"], parameters["θ_Λ"])
+end
+
+### ---------------------------------------------- ###
+### ----------------- INFERENCE ------------------ ###
+### ---------------------------------------------- ###
+function run_inference(state, parameters, data)
+
+    # Add missing values to the observations to match the horizon
+    observations = vcat(convert.(Float64, data["observation"]), [missing for _ in 1:state["horizon"]])
+
+    inference_results = infer(
+        model = AR_model(order = state["order"], parameters = parameters, state = state),
+        data = (y = UnfactorizedData(observations),),
+        meta = AR_meta(state["order"]),
+        constraints = AR_constraints(),
+        initialization = AR_init(parameters),
+        options = (limit_stack_depth = 300,),
+        iterations = 20,
+        returnvars = KeepLast()
+    )
+
+    result = Dict("states" => inference_results.posteriors[:x])
+    state["x_μ"] = mean(last(inference_results.posteriors[:x]))
```

Comment:
I think you want to take `first` instead of `last`?

Reply:
you are right, this doesn't make any sense whatsoever